Realm/RLMSyncPermissionValue.mm

```diff
@@ -76,27 +78,41 @@ - (instancetype)initWithRealmPath:(NSString *)path
         _accessLevel = accessLevel;
         _path = path;
         _userID = userID;
+        if (!_userID) {
```

Comment:
What's the rationale for doing a runtime check for just this parameter? Both strings are declared as nonnull, so I would expect either both or neither to be checked.

I'd also be inclined to write this as argument validation done before the init stuff rather than validation of the object's initialized state, but I won't try to pretend that's anything other than personal preference.